src/diffusers/models/attention.py

```diff
@@ -284,22 +284,52 @@ def forward(self, hidden_states):
         key_proj = self.key(hidden_states)
         value_proj = self.value(hidden_states)
 
-        # transpose
-        query_states = self.transpose_for_scores(query_proj)
-        key_states = self.transpose_for_scores(key_proj)
-        value_states = self.transpose_for_scores(value_proj)
+        if self.num_heads > 1:
+            query_states = self.transpose_for_scores(query_proj)
+            key_states = self.transpose_for_scores(key_proj)
+            value_states = self.transpose_for_scores(value_proj)
+        else:
+            query_states, key_states, value_states = query_proj, key_proj, value_proj
 
         # get scores
-        scale = 1 / math.sqrt(math.sqrt(self.channels / self.num_heads))
-        attention_scores = torch.matmul(query_states * scale, key_states.transpose(-1, -2) * scale)  # TODO: use baddmm
+        scale = 1 / math.sqrt(self.channels / self.num_heads)
```

Comment:
Could me move this line before the first `if-else` statement and then have only one big if-else statement? IMO this would make the code more readable

Reply:
done, rebased, force-pushed.

there's still another if-else below. merging that is possible, but only by duplicating the softmax.